aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -821,25 +821,42 @@ TORCH_IMPL_FUNC(index_add_cpu_out)
 
     // explicitly capture all required variables to work around windows build
     // TODO: fix this when windows can correctly capture variables in nested lambda
-    AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(ScalarType::Half, ScalarType::Bool, ScalarType::BFloat16,
-      result.scalar_type(), "index_add_", [&result, &source, &dim, &index_contig, &numel, &alpha] {
-      auto alpha_value = alpha.to<scalar_t>();
-      auto result_stride = result.dim() == 0 ? 1 : result.stride(dim);
-      auto source_stride = source.dim() == 0 ? 1 : source.stride(dim);
-      // TODO: Maybe TensorAccessor can be used here?
-      auto* result_ptr = result.data_ptr<scalar_t>();
-      auto* source_ptr = source.data_ptr<scalar_t>();
-      AT_DISPATCH_INDEX_TYPES(index_contig.scalar_type(), "index_add_cpu_",
-        [&index_contig, &numel, &result, &result_ptr, &result_stride, &source_ptr, &source_stride, &alpha_value] {
-        auto index_data = index_contig.data_ptr<index_t>();
-        for (const auto i : c10::irange(numel)) {
-            auto self_i = index_data[i];
-            TORCH_CHECK_INDEX((self_i >= 0) && (self_i < result.numel()), "index out of range in self");
-            scalar_t *self_ip = result_ptr + self_i * result_stride;
-            *self_ip += *(source_ptr + i * source_stride) * alpha_value;
-        }
-      });
-    });
+    AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND4(
+        ScalarType::Half,
+        ScalarType::Bool,
+        ScalarType::BFloat16,
+        ScalarType::ComplexHalf,
```

Comment:
I can cherry pick this commit and move it to a separate PR, but it was such a small change.

Reply:
I know, but from a release process perspective it'll have to be a separate PR. Luckily GH1 makes this quick to merge. I'll stamp it as soon as it arrives. I'd also leave out the code formatting so it's truly just a single line change.

Reply:
I'd leave the change in this PR, create a separate, unstacked PR with this change only and then rebase once it's merged into master. That way you're not blocked and we don't have to change this current PR.

Reply:
Done, see #79897.